### ISSUE TITLE
ERROR BUG in /usr/share/rear/lib/layout-functions.sh:

### ISSUE THREAD

**Issue by techadmin7** [rear/rear#1717]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* rear version (/usr/sbin/rear -V): Relax-and-Recover 2.00 / Git

* OS version (cat /etc/rear/os.conf or lsb_release -a): 
OS_VENDOR=RedHatEnterpriseServer
OS_VERSION=7

* rear configuration files (cat /etc/rear/site.conf or cat /etc/rear/local.conf):
[default.conf.txt](https://github.com/rear/rear/files/1678146/default.conf.txt)

* Are you using legacy BIOS or UEFI boot? UEFI 

* Brief description of the issue:
```
ERROR: 
====================
BUG in /usr/share/rear/lib/layout-functions.sh:
'Could not determine size of disk sda/sda1, please file a bug.'
--------------------
Please report this issue at https://github.com/rear/rear/issues
and include the relevant parts from /var/log/rear/rear-huey.log.lockless
preferably with full debug information via 'rear -d -D checklayout'
====================
Aborting due to an error, check /var/log/rear/rear-huey.log.lockless for details
/bin/sh: line 1: 56687 Terminated              /usr/sbin/rear checklayout"
```
* Work-around, if any: N/A

**Comment by gozora** [comment]
HI, 
Maybe you could upgrade to ReaR >= 2.2.
There was https://github.com/rear/rear/pull/1418 implemented which might solve your issue.

V.

**Comment by techadmin7** [comment]
I understand updating however I have two other servers same build model same everything that isn’t having this problem at all.

**Comment by gozora** [comment]
This is a captious one, I had 8 machines and only 2 were affected (but not always).

V.

**Comment by techadmin7** [comment]
Ok. Is the best way to update to run “yum update pkgname” ??

**Comment by techadmin7** [comment]
[root@huey bin]# cat /var/log/rear/rear-huey.log.lockless
2018-01-30 01:30:01 Relax-and-Recover 2.00 / Git
2018-01-30 01:30:01 Command line options: /usr/sbin/rear checklayout
2018-01-30 01:30:01 Using log file: /var/log/rear/rear-huey.log.lockless
2018-01-30 01:30:01 Including /etc/rear/os.conf
2018-01-30 01:30:01 Including conf/Linux-i386.conf
2018-01-30 01:30:01 Including conf/GNU/Linux.conf
2018-01-30 01:30:01 Including /etc/rear/local.conf
2018-01-30 01:30:01 Running 'init' stage
2018-01-30 01:30:01 Including init/default/010_set_drlm_env.sh
2018-01-30 01:30:01 Including init/default/030_update_recovery_system.sh
2018-01-30 01:30:01 Including init/default/050_check_rear-recover_mode.sh
2018-01-30 01:30:01 Finished running 'init' stage in 0 seconds
2018-01-30 01:30:01 Using build area '/tmp/rear.zsH0etc0NUinzRI'
2018-01-30 01:30:01 Running checklayout workflow
2018-01-30 01:30:01 Running 'layout/precompare' stage
2018-01-30 01:30:01 Including layout/precompare/default/110_check_layout_file.sh
2018-01-30 01:30:01 Finished running 'layout/precompare' stage in 0 seconds
2018-01-30 01:30:01 Running 'layout/save' stage
2018-01-30 01:30:01 Including layout/save/GNU/Linux/100_create_layout_file.sh
2018-01-30 01:30:01 Creating disk layout
2018-01-30 01:30:01 Preparing layout directory.
2018-01-30 01:30:01 Including layout/save/GNU/Linux/150_save_diskbyid_mappings.sh
2018-01-30 01:30:01 Saved diskbyid_mappings
2018-01-30 01:30:01 Including layout/save/GNU/Linux/200_partition_layout.sh
2018-01-30 01:30:01 Saving disk partitions.
2018-01-30 01:30:01 ERROR:
====================
BUG in /usr/share/rear/lib/layout-functions.sh:
'Could not determine size of disk sda/sda1, please file a bug.'
--------------------
Please report this issue at https://github.com/rear/rear/issues
and include the relevant parts from /var/log/rear/rear-huey.log.lockless
preferably with full debug information via 'rear -d -D checklayout'
====================
==== Stack trace ====
Trace 0: /usr/sbin/rear:504 main
Trace 1: /usr/share/rear/lib/checklayout-workflow.sh:18 WORKFLOW_checklayout
Trace 2: /usr/share/rear/lib/framework-functions.sh:85 SourceStage
Trace 3: /usr/share/rear/lib/framework-functions.sh:45 Source
Trace 4: /usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh:269 source
Trace 5: /usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh:63 extract_partitions
Trace 6: /usr/share/rear/lib/layout-functions.sh:534 get_disk_size
Trace 7: /usr/share/rear/lib/_input-output-functions.sh:166 BugIfError
Trace 8: /usr/share/rear/lib/_input-output-functions.sh:160 BugError
Message:
====================
BUG in /usr/share/rear/lib/layout-functions.sh:
'Could not determine size of disk sda/sda1, please file a bug.'
--------------------
Please report this issue at https://github.com/rear/rear/issues
and include the relevant parts from /var/log/rear/rear-huey.log.lockless
preferably with full debug information via 'rear -d -D checklayout'
====================
== End stack trace ==
Error: /dev/sdc: unrecognised disk label
Error: /dev/sdd: unrecognised disk label
2018-01-30 01:30:02 Running exit tasks.
2018-01-30 01:30:02 Finished in 1 seconds
2018-01-30 01:30:02 Removing build area /tmp/rear.zsH0etc0NUinzRI
2018-01-30 01:30:02 End of program reached

**Comment by gdha** [comment]
@techadmin7 Do you have a support contract with RedHat? If yes, please open an incident for this with RH so that they can fix their ReaR version too.
If you want to test the latest stable version of rear go to our Download page and select the proper OS -
http://download.opensuse.org/repositories/Archiving:/Backup:/Rear/RHEL_7/x86_64/

**Comment by jsmeix** [comment]
@techadmin7 
we cannot tell you what the best way is how to upgrade
a particular software on your particular Linux distribution.
For such kind of Linux distribution specific questions
you may better ask at your particular Linux distribution.

In general regarding ReaR version upgrades see the section
"Version upgrades with Relax-and-Recover" at
https://en.opensuse.org/SDB:Disaster_Recovery

In general regarding debugging issues in ReaR see the section
"Debugging issues with Relax-and-Recover" at
https://en.opensuse.org/SDB:Disaster_Recovery

FYI
how you could test the current ReaR GitHub master code
without get things messed up with an already installed
ReaR software package:

Basically "git clone" it into a directory and then
configure and run ReaR from within that directory like:
<pre>
# git clone https://github.com/rear/rear.git

# cd rear

# vi etc/rear/local.conf

# usr/sbin/rear -D mkbackup
</pre>
Note the relative paths "etc/rear/" and "usr/sbin/".

When the issue still appears with the current ReaR GitHub master code
provide your complete etc/rear/local.conf and a "rear -D mkbackup" log file, cf.
https://en.opensuse.org/SDB:Disaster_Recovery#Debugging_issues_with_Relax-and-Recover

**Comment by techadmin7** [comment]
@gdha I do have a contract with Redhat. I have opened a case just waiting for there response. Thank you for your advise.
